frontend/src/metabase/visualizations/components/Visualization.jsx

```diff
@@ -296,17 +300,21 @@ class Visualization extends React.PureComponent {
     }
   };
 
+  /* eslint-disable complexity */
```

Comment:
We have a max complexity of value in the app of 54 because of this specific function. In combination with my comment below regarding the logic around hiding the header, the complexity is only bumped slightly higher.

Obviously not the greatest solution but if need be we can bump the number found here (https://github.com/metabase/metabase/blob/c55faa6cb69ab71387d5428d28c7f5541e18a73f/.eslintrc#L35) so that it matches the complexity of this function.

Another simple way to reduce it might be pulling out the logic that sets `hasHeader` into a separate predicate function. Up to you.

Reply:
The header computation stuff has a ton of dependencies in the function so it's tricky to pull out. But I pulled out the `loading` computation into a separate `isLoading` function and that was enough to get the complexity below 54.